source/GameInterface/Services/Buildings/Messages/BuildingCreated.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Common.Messaging;
+using GameInterface.Services.Armies.Data;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.CampaignSystem.Party;
+using TaleWorlds.CampaignSystem.Settlements;
+using TaleWorlds.CampaignSystem.Settlements.Buildings;
+
+namespace GameInterface.Services.Armies.Messages.Lifetime;
+
+/// <summary>
+/// Event to create a new building.
+/// </summary>
+public record BuildingCreated : IEvent
+{
+    public Building Building { get; }
+    public BuildingType BuildingType { get; }
+    public Town Town { get; }
+    public float BuildingProgress { get; }
+    public int CurrentLevel { get; }
+
+    public BuildingCreated(Building building, BuildingType buildingType, Town town, float buildingProgress, int currentLevel)
+    {
+        Building = building;
+        BuildingType = buildingType;
+        Town = town;
+        BuildingProgress = buildingProgress;
+        CurrentLevel = currentLevel;
+    }
```

Comment:
Since we are not going to use the constructor we don't need those fields

```suggestion
    public Building Building { get; }

    public BuildingCreated(Building building)
    {
        Building = building;
    }
```